tests/lax_numpy_test.py

```diff
@@ -2090,6 +2090,23 @@ def testTrilIndicesFrom(self, shape, dtype, k):
     args_maker = lambda: [rng(shape, dtype), k]
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
 
+  @jtu.sample_product(
+    dtype=default_dtypes,
+    shape=[(0, 0), (0, 1), (1, 0), (1, 1), (2, 0), (2, 1), (2, 2), (1, 2), (0, 2), (2, 3), (2, 2, 2), (2, 2, 2, 2)],
+  )
+  def testFillDiagonal(self, dtype, shape):
+    rng = jtu.rand_default(self.rng())
+
+    def np_fun(a, val):
+      a_copy = a.copy()
+      np.fill_diagonal(a_copy, val)
+      return a_copy
+
+    jnp_fun = partial(jnp.fill_diagonal, inplace=False)
+    args_maker = lambda : [rng(shape, dtype), rng((), dtype)]
```

Comment:
One last detail: I think `np.fill_diagonal` allows for non-scalar arguments to `val`. We should either add test coverage for that, or raise NotImplementedError in that case.